forui/lib/src/widgets/avatar/avatar_content.dart

```diff
@@ -0,0 +1,79 @@
+part of 'avatar.dart';
+
+class _AvatarContent extends StatelessWidget {
+  final FAvatarStyle? style;
+  final ImageProvider image;
+  final double size;
+  final Widget? placeholder;
+
+  const _AvatarContent({
+    required this.image,
+    required this.style,
+    required this.size,
+    this.placeholder,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final style = this.style ?? context.theme.avatarStyle;
+
+    final placeholder = this.placeholder ?? _IconPlaceholder(style: style, size: size);
+
+    return Image(
+      height: size,
+      width: size,
+      filterQuality: FilterQuality.medium,
+      image: image,
+      errorBuilder: (context, exception, stacktrace) => placeholder,
+      frameBuilder: (context, child, frame, wasSynchronouslyLoaded) {
+        if (wasSynchronouslyLoaded) {
+          return child;
+        }
+        return AnimatedSwitcher(
+          duration: const Duration(milliseconds: 500),
+          child: frame == null ? placeholder : child,
+        );
+      },
+      loadingBuilder: (context, child, loadingProgress) {
+        if (loadingProgress == null) {
+          return child;
+        }
+        return placeholder;
+      },
+      fit: BoxFit.cover,
+    );
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('style', style))
+      ..add(DoubleProperty('size', size))
+      ..add(DiagnosticsProperty('image', image));
+  }
+}
+
+class _IconPlaceholder extends StatelessWidget {
```

Comment:
If the above is updated to `fallback` it could make sense to just name this `_Placeholder`. Since this is truly the placeholder if no fallback/child is provided.